### PR TITLE
Add diagram of how transport actions are checked

### PR DESCRIPTION
**Pull request by peternied**
### Description
Add diagram of how transport actions are checked onto the readme.  This diagram is much easier to see in 'rich' diff mode of the readme.

### Check List
- [ ] New functionality includes testing
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cwperks** [comment]
Thank you for adding this @peternied. Do you think it would also be possible to add the relevant extension points that the SecurityPlugin utilizes? `getRestHandlerWrapper()` to wrap RestRequests to authenticate and enrich the threadcontext with user info and also `getActionFilters()` to implement authorization?

**Review by scrawfor99 — APPROVED** [review]
Great job Peter! This looks very good to me. I left one comment but also granted approval so that if you clarify, you don't have to wait for my re-review.

File: README.md
Comment (scrawfor99) [inline README.md:141]: Should this be above line 140? Wouldn't we want to check the action is allowed before executing the request outside the user context?

Comment (peternied) [reply README.md]: Tweaked this, it should be a check within the security plugin, similar to the earlier one

**Review by cwperks — APPROVED** [review]

**Review by scrawfor99 — APPROVED** [review]
Looks great!

**Review by DarshitChanpura — APPROVED** [review]

**Comment by DarshitChanpura** [comment]
@peternied CI failures should be fixed once latest changes from main are merged in this branch.

**Pull request merged**
